src/librustc_lint/array_into_iter.rs

```diff
@@ -48,14 +48,31 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for ArrayIntoIter {
             // argument.
             let receiver_arg = &args[0];
 
-            // Test if the original `self` type is an array type.
-            match cx.tables.expr_ty(receiver_arg).kind {
-                ty::Array(..) => {}
-                _ => return,
+
+            fn is_array(ty: &ty::TyS<'_>) -> bool {
```

Comment:
I think this is useful elsewhere, so add this as a method on `ty`. :)

Reply:
I think I'd go with the `matches` version (mentioned below) and remove my helper function. Or would you really like to have the `is_array` method on `TyS`? I also thought about adding it, but one could also add one `is_*` method for each variant of `TyKind` then... which is a lot.